src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -56,11 +56,34 @@ export class ShowCondition {
     }
   }
 
-  private findValueForComplexCondition(fields: any, head: string, tail: string[]) {
+  private findValueForComplexCondition(fields: any, head: string, tail: string[], path?: string) {
```

Comment:
This method is a bit bulky. Is it possible to break it down a bit?

Reply:
Agree with @andrewfolga